pkg/clients/aws/client_test.go

```diff
@@ -0,0 +1,94 @@
+package aws
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const fakeAccountNum = "123456789101"
+
+func TestGetRancherLicense(t *testing.T) {
+	tests := []struct {
+		name              string // name of the test, to be displayed on failure
+		hasAmeaLicense    bool   // if the account has a license for the AMEA product sku
+		hasNonAmeaLicense bool   // if the account has a licensed for the non AMEA product sku
+		includeProductSku bool   // if the return from aws should include or exclude a product sku
+		desiredLicense    string // which license our client should pick - amea, nona-amea, or nothing
+		errDesired        bool   // if we wanted an error for this test case
+	}{
+		{
+			name:              "test amea license",
+			hasAmeaLicense:    true,
+			hasNonAmeaLicense: false,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test non-amea license",
+			hasAmeaLicense:    false,
+			hasNonAmeaLicense: true,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUNonAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test amea + non-amea license - should not occur in reality",
+			hasAmeaLicense:    true,
+			hasNonAmeaLicense: true,
+			includeProductSku: true,
+			desiredLicense:    rancherProductSKUAmea,
+			errDesired:        false,
+		},
+		{
+			name:              "test no valid license",
+			hasAmeaLicense:    false,
+			hasNonAmeaLicense: false,
+			includeProductSku: false,
+			desiredLicense:    "",
+			errDesired:        true,
+		},
+		{
+			name:              "test no product sku amea",
```

Comment:
I'm unclear why this is a valid test. Shouldn't https://github.com/rancher/csp-adapter/pull/21/files#diff-a9176a0f4867c18136a210d479eca53dc199a64eb2725813e213c69a944d6a50R136-R139 guarantee this doesn't happen? Why would there be a valid amea license if there's no product sku?

Reply:
You are correct about the purpose of that code. Since the license that we get from AWS uses a `*string` as the type for the product sku, it is conceivable that AWS won't return this value for us (i.e that it could be nil). However, this PR did change the code for `CheckoutRancherLicense` to use the product sku from the license that we got. Because of this, I need to make sure (somewhere) that this value is being set. I chose to do that in `GetRancherLicense`, and because of that, added a test to make sure that even if AWS doesn't return the SKU, it has still been set on the license.

Reply:
I think if we're getting a license back but it doesn't have the SKU set, that should be treated as an error. If it should never happen, then if it does it is invalid. If we just set the value directly, we're setting it on a license whose state we don't really understand. Consider if `getLicenseForProductID` is called for Amea, an invalid license is returned, but calling it for NonAmea would have returned a valid license. The SKU would still be set as Amea on this invalid license.

Reply:
I get where you are coming from, but I don't think I quite agree. AWS docs (https://docs.aws.amazon.com/license-manager/latest/APIReference/API_GrantedLicense.html) indicate that this value is optional. In testing, I haven't seen it come back missing, but given that the API doesn't guarantee it.

Would you be ok with me leaving the test/code as is but removing the comment? I think the comment is by far the most inaccurate part of this.

Reply:
I've revised the comment, please LMK if this doesn't fit with what you were thinking.